qa/vagrant/src/main/java/org/elasticsearch/packaging/test/ArchiveTestCase.java

```diff
@@ -192,7 +194,70 @@ public void test52BundledJdkRemoved() throws IOException {
         }
     }
 
-    public void test60AutoCreateKeystore() {
+    public void test53JavaHomeWithSpecialCharacters() throws Exception {
+        assumeThat(installation, is(notNullValue()));
+
+        Platforms.onWindows(() -> {
+            final Shell sh = new Shell();
+            final String originalPath = sh.run("$Env:PATH").stdout.trim();
+            final String javaHome = sh.run("$Env:JAVA_HOME").stdout.trim();
+
+            try {
+                // once windows 2012 is no longer supported and powershell 5.0 is always available we can change this command
+                sh.runIgnoreExitCode("cmd /c mklink /D 'C:\\Program Files (x86)\\java' $Env:JAVA_HOME");
+
+                sh.getEnv().put("JAVA_HOME", "C:\\Program Files (x86)\\java");
+
+                //verify ES can start, stop and run plugin list
+                Archives.runElasticsearch(installation, sh);
+
+                Archives.stopElasticsearch(installation);
+
+                String pluginListCommand = installation.bin + "/elasticsearch-plugin list";
+                Result result = sh.run(pluginListCommand);
+                assertThat(result.exitCode, equalTo(0));
+
+            } finally {
+                //clean up sym link
+                sh.runIgnoreExitCode("cmd /c del /F /Q 'C:\\Program Files (x86)\\java' ");
+                sh.getEnv().put("JAVA_HOME", javaHome);
+            }
+        });
+
+        Platforms.onLinux(() -> {
+            final Shell sh = new Shell();
+            final String javaHome = sh.run("echo $JAVA_HOME").stdout.trim();
+
+            try {
+
+                // Create temporary directory with a space and link to java binary.
+                // Use it as java_home
+                final String temp = sh.runIgnoreExitCode("mktemp -d --suffix=\"java home\"").stdout.trim();
+                final String java = sh.runIgnoreExitCode("which java").stdout.trim();
```

Comment:
Look for $SYSTEM_JAVA_HOME. See other tests that set JAVA_HOME. We no longer use java from the path.